libkbfs/folder_branch_ops.go

```diff
@@ -574,6 +598,16 @@ func (fbo *folderBranchOps) setHeadLocked(
 	if !isFirstHead {
 		wasReadable = fbo.head.IsReadable()
 
+		if !fbo.headTrusted {
+			err := fbo.validateHead(ctx, md)
```

Comment:
Looks like `trusted` is only ever false when setting the first head, so I think we want to make sure that `trusted` is true in the `!isFirstHead` clause.